### PR TITLE
ENH: Add support for params, query_schema, and sql in PySpark backend

### PR DESCRIPTION
**Pull request by icexelloss**
Pass ibis/all/test_client.py and ibis/all/test_params.py

test_params.py is partially supported

### CODE REVIEW

**Comment by icexelloss** [comment]
cc @hjoo @toryhaavik

**Comment by icexelloss** [comment]
Gentle ping @toryhaavik

File: ibis/pyspark/client.py [hunk 1]
```diff
@@ -12,26 +12,37 @@ class PySparkClient(SparkClient):
     An ibis client that uses PySpark SQL Dataframe
     """
 
+    dialet = PySparkDialect
```
Comment (toryhaavik) [inline ibis/pyspark/client.py]: typo, should be `dialect`

File: ibis/pyspark/api.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 from ibis.pyspark.client import PySparkClient
+from ibis.pyspark.compiler import dialect  # noqa: F401
```
Comment (toryhaavik) [inline ibis/pyspark/api.py]: why does this get put in the public api?

Comment (icexelloss) [reply ibis/pyspark/api.py]: It is because this line here:
https://github.com/ibis-project/ibis/blob/master/ibis/tests/backends.py#L145

**Pull request merged**